src/tests/functions/gif/test_gif.py

```diff
@@ -52,3 +64,50 @@ def test_gif_empty_params():
     result = gif(request_mock)
 
     assert result == DEFAULT_GIF
+
+
+def test_success_200():
+    """Test success api call"""
+    api_request_json = """{"results": [
+                {
+                    "media_formats": {
+                    "gif": {
+                        "url": "https://c.tenor.com/HvtIOMNuwhYAAAAC/wat-what.gif",
+                        "duration": 0,
+                        "preview": "",
+                        "dims": [
+                        436,
+                        498
+                        ],
+                        "size": 7986816
+                    }
+                    }
+                }
+                ]
+            }"""
+
+    result = "https://c.tenor.com/HvtIOMNuwhYAAAAC/wat-what.gif"
+
+    api_status = 200
+
+    assert result == get_gif_from_api(api_status, api_request_json)
+
+
+def test_success_200_bad_json():
+    """Test bad json format api"""
+    api_request_json = """{ }"""
+
+    result = "https://tenor.com/view/im-dead-vanilla-patay-na-ako-dead-nako-patay-ako-gif-22020482"
+
+    api_status = 200
+
+    assert result == get_gif_from_api(api_status, api_request_json)
+
+
+def test_fail_not_200():
+    """Test API request failure"""
+    api_request_json = """{ }"""
+
+    api_status = 503
```

Comment:
This api status might break the main node if return to it
We want the commands to return code 200 for now

We will add error handling in the main node in the future